### PR TITLE
Add `cors` middleware helper and `all` route handler

### LINKED ISSUE #10: Add CORS helper
[Similar to Hono](https://hono.dev/docs/middleware/builtin/cors), we should add the ability to configure CORS.

### PR DESCRIPTION
**Pull request by khairul169**
This PR add these features:
- `cors` middleware helper
- `all` route handler to catch any HTTP methods

fixes #10 

--

anyway, thank you so much for the project, i really like the idea. gonna use this library for my future project! :D

### CODE REVIEW

File: src/Router.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ public function findRoute(ServerRequestInterface $request): ?array
 
         foreach ($this->routes as $route) {
             if (
-                $route["method"] === $method &&
+                ($route["method"] === $method || $route["method"] === "ALL" || $method === "OPTIONS") &&
```
Comment (khairul169) [inline src/Router.php]: I'm not sure if this is okay; any feedback would be appreciated. :)

**Comment by notrab** [comment]
This is really awesome @khairul169, fantastic work!

Let me test this out locally and see if there's any edge cases. 

I really should add some tests as well now this is a bit more serious heh 😅

**Comment by notrab** [comment]
Hey @khairul169, I took another look at this but I won't have time today to apply the changes directly to get this merged.

I'm out of the office this weekend, but I will resume on Monday and get this shipped 🚀 

A few things I wanted to do:

1. Rebase with `main`
2. Abstract methods into a private array that can be shared, as they're defined a few times now
3. Use camelCase instead of snake_case for options (or is snake_case preferred in PHP?)
4. Add support for multiple origins as an array or a callback function
5. Implement the `allowHeaders` functionality similar to Hono
6. Add more PHPDoc comments
7. Write a test

**Comment by notrab** [comment]
@khairul169 I pushed some changes. I'd love it if you could test it out, and make sure it makes sense!

**Comment by notrab** [comment]
I'm going to merge this now and if there's any changes required, we can update. Thanks for your work on this @khairul169 ❤️
